quinn-proto/src/streams.rs

```diff
@@ -201,10 +202,17 @@ impl Streams {
         id.index() >= self.next[id.dir() as usize]
     }
 
-    fn insert(&mut self, remote: bool, id: StreamId) {
+    fn insert(&mut self, params: Option<&TransportParameters>, remote: bool, id: StreamId) {
         let bi = id.dir() == Dir::Bi;
         if bi || !remote {
-            assert!(self.send.insert(id, Send::new()).is_none());
+            let max_data = params.map_or(0, |params| match id.dir() {
+                Dir::Uni => params.initial_max_stream_data_uni,
+                // Remote/local appear reversed here because the transport parameters are named from
+                // the perspective of the peer.
+                Dir::Bi if remote => params.initial_max_stream_data_bidi_local,
```

Comment:
You seem to have `initial_max_stream_data_bidi_local` twice here. Probably not intended (especially given the comment)?

Reply:
Whoops, good catch!